src/gravitational_waves/gw.cpp

```diff
@@ -220,61 +331,391 @@ Nintegrate_SNR(GravitationalWave &obj, const double fmin, const double fmax)
+  const double E0V = BACalc.GetBounceSol(Tstar) * Tstar / 2.;
+  // Initial bubble radius
+  double R0 = pow(3 * E0V / (4 * M_PI * dV), 1. / 3.);
+
+  // Mean bubble seperation
+  const double Rstar = HR / BACalc.HubbleRate(Tstar);
+  // Energy densitity radiation dominated Universe
+  const double RhoGamma = BACalc.CalculateRhoGamma(Tstar);
+
+  // Calculate masses squared in false and true vacuum
+  std::vector<double> HiggsSq_False =
+      BACalc.modelPointer->HiggsMassesSquared(FalseVacuum, Tstar);
+  std::vector<double> LeptonSq_False =
+      BACalc.modelPointer->LeptonMassesSquared(FalseVacuum);
+  std::vector<double> QuarkSq_False =
+      BACalc.modelPointer->QuarkMassesSquared(FalseVacuum);
+  std::vector<double> GaugeSq_False =
+      BACalc.modelPointer->GaugeMassesSquared(FalseVacuum, Tstar);
+
+  std::vector<double> HiggsSq_True =
+      BACalc.modelPointer->HiggsMassesSquared(TrueVacuum, Tstar);
+  std::vector<double> LeptonSq_True =
+      BACalc.modelPointer->LeptonMassesSquared(TrueVacuum);
+  std::vector<double> QuarkSq_True =
+      BACalc.modelPointer->QuarkMassesSquared(TrueVacuum);
+  std::vector<double> GaugeSq_True =
+      BACalc.modelPointer->GaugeMassesSquared(TrueVacuum, Tstar);
+
+  double P_LO = 0; // Pressure at LO. 1 -> 1
+
+  for (auto massSq : HiggsSq_True)
+    P_LO += massSq;
+  for (auto massSq : GaugeSq_True)
+    P_LO += 3 * massSq; // 3 from polarization
+  for (auto massSq : LeptonSq_True)
+    P_LO += massSq / 2.;
+  for (auto massSq : QuarkSq_True)
+    P_LO += 3 * massSq / 2.; // 3 from colour
+  for (auto massSq : HiggsSq_False)
+    P_LO -= massSq;
+  for (auto massSq : GaugeSq_False)
+    P_LO -= 3 * massSq; // 3 from polarization
+  for (auto massSq : LeptonSq_False)
+    P_LO -= massSq / 2.;
+  for (auto massSq : QuarkSq_False)
+    P_LO -= 3 * massSq / 2.; // 3 from colour
+
+  P_LO *= pow(Tstar, 2) / 24.; // pressure LO normalization
+
+  // Pressure at NLO. 1 -> N
+  // This only works for the SU(2)_L x U(1)_Y gauge group. But can be easily
+  // generalized.
+  const double W_coupling = BACalc.modelPointer->SMConstants.C_g / sqrt(2);
+  const double Z_coupling =
+      BACalc.modelPointer->SMConstants.C_g /
+      sqrt(1. - BACalc.modelPointer->SMConstants.C_sinsquaredWeinberg);
+  const double A_coupling =
+      BACalc.modelPointer->SMConstants.C_g *
+      sqrt(BACalc.modelPointer->SMConstants.C_sinsquaredWeinberg);
+  double P_NLO = 0;
+  if (pnlo_scaling == 1)
+  {
+    // Mass matrix is ordered
+    P_NLO += GaugeSq_True.at(0) * A_coupling;
+    P_NLO += 2 * GaugeSq_True.at(1) * W_coupling;
+    P_NLO += GaugeSq_True.at(3) * Z_coupling;
+
+    P_NLO -= GaugeSq_False.at(0) * A_coupling;
+    P_NLO -= 2 * GaugeSq_False.at(1) * W_coupling;
+    P_NLO -= GaugeSq_False.at(3) * Z_coupling;
+  }
+  else if (pnlo_scaling == 2)
+  {
+    P_NLO = 2 * pow(W_coupling, 2) + pow(Z_coupling, 2) + pow(A_coupling, 2);
+    P_NLO *= pow(Tstar, 4); // pressure NLO normalization
```

Comment:
```suggestion
    P_NLO *= pow(Tstar, 3); // pressure NLO normalization
```
https://arxiv.org/pdf/2412.02645 and https://arxiv.org/pdf/2210.07075 have `Tp^3` here, not `Tp^4`?

Reply:
True, you are right, for $\gamma^1$, the scaling is $T_{\*}^{3}$, you were correct. But isn't the $T_*^4$ factor missing in case `pnlo_scaling == 1`?